src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -1113,6 +1116,79 @@ bool isLambdaConversion(BoundExpression expr)
             { WasCompilerGenerated = true };
         }
 
+        private BoundExpression CreateParamsSpan(
+            SyntaxNode syntax,
+            TypeSymbol paramArrayType,
+            ImmutableArray<BoundExpression> arrayArgs)
+        {
+            var elementType = paramArrayType.GetParamsElementType().Type;
+            int length = arrayArgs.Length;
+
+            // PROTOTYPE: If we actually allocate from the stack, it may be necessary to
+            // clear the execution stack before allocation. (See corresponding requirement
+            // for 'stackalloc' in LocalRewriter.VisitStackAllocArrayCreationBase().)
+
+            // For now, simply allocate the array on the heap: new Span<T>(new T[length])
+            var intType = _compilation.GetSpecialType(SpecialType.System_Int32);
+            var array = new BoundArrayCreation(
+                syntax,
+                ImmutableArray.Create(MakeLiteral(syntax, ConstantValue.Create(length), intType, this)),
+                initializerOpt: null,
+                ArrayTypeSymbol.CreateSZArray(_compilation.SourceAssembly, TypeWithAnnotations.Create(elementType)))
+            { WasCompilerGenerated = true };
+
+            MethodSymbol spanConstructor;
+            MethodSymbol spanGetItem;
+            if (!TryGetWellKnownTypeMember(syntax, WellKnownMember.System_Span_T__ctorArray, out spanConstructor) ||
+                !TryGetWellKnownTypeMember(syntax, WellKnownMember.System_Span_T__get_Item, out spanGetItem))
+            {
+                return BadExpression(syntax, paramArrayType, ImmutableArray<BoundExpression>.Empty);
+            }
+
+            var spanType = spanConstructor.ContainingType.Construct(elementType);
+            spanConstructor = spanConstructor.AsMember(spanType);
+            spanGetItem = spanGetItem.AsMember(spanType);
+
+            // PROTOTYPE: Test use-site diagnostics.
+            var useSiteInfo = new CompoundUseSiteInfo<AssemblySymbol>(_diagnostics, _compilation.Assembly);
+            var conversion = _compilation.Conversions.ClassifyImplicitConversionFromType(spanType, paramArrayType, ref useSiteInfo);
+            conversion.MarkUnderlyingConversionsCheckedRecursive();
+            _diagnostics.Add(syntax, useSiteInfo);
+
+            if (!conversion.IsValid)
+            {
+                _diagnostics.Add(ErrorCode.ERR_NoImplicitConv, syntax.Location, spanType, paramArrayType);
+                return BadExpression(syntax, paramArrayType, ImmutableArray<BoundExpression>.Empty);
+            }
+
+            var sideEffects = ArrayBuilder<BoundExpression>.GetInstance();
+            var span = _factory.New(spanConstructor, array);
+            var temp = _factory.StoreToTemp(span, out var assignment);
+            sideEffects.Add(assignment);
+
+            // temp[0] = arrayArgs[0];
+            // ...
+            // temp[n - 1] = arrayArgs[n - 1];
+            // temp
+            for (int i = 0; i < arrayArgs.Length; i++)
```

Comment:
>for (int i = 0; i < arrayArgs.Length; i++) [](http://example.com/codeflow?start=12&length=42)

I am not sure why go through all this trouble when we can initialize the allocated array directly and we already have a helper for that.

Reply:
To allow for allocation to be handled by a runtime helper method.